pkg/webhook/validation/uservalidation.go

```diff
@@ -158,19 +159,22 @@ func checkCRDGroup(group string) bool {
 }
 
 // ValidateMCIdentity returns admission allowed/denied based on the member cluster's identity.
-func ValidateMCIdentity(ctx context.Context, client client.Client, userInfo authenticationv1.UserInfo, resourceNamespacedName types.NamespacedName, resourceKind, mcName string) admission.Response {
+func ValidateMCIdentity(ctx context.Context, client client.Client, userInfo authenticationv1.UserInfo, resourceNamespacedName types.NamespacedName, resourceKind, subResource, mcName string) admission.Response {
 	var mc fleetv1alpha1.MemberCluster
 	if err := client.Get(ctx, types.NamespacedName{Name: mcName}, &mc); err != nil {
 		// fail open, if the webhook cannot get member cluster resources we don't block the request.
 		klog.V(2).ErrorS(err, fmt.Sprintf("failed to get member cluster resource for request to modify %s, allowing request to be handled by api server", resourceKind),
 			"user", userInfo.Username, "groups", userInfo.Groups, "namespacedName", resourceNamespacedName)
-		return admission.Allowed(fmt.Sprintf(resourceAllowedGetMCFailed, userInfo.Username, userInfo.Groups, resourceKind, resourceNamespacedName))
+		if apierrors.IsNotFound(err) {
+			return admission.Errored(http.StatusNotFound, err)
+		}
+		return admission.Errored(http.StatusInternalServerError, err)
```

Comment:
why change from fail open to fail close?

Reply:
I went a rabbit hole to figure out how to use admission.Errored and missed that we are using fail open for this case my bad will revert